Document process for stabilizing a syscall driver (#4329)
tock/tock: doc/Maintenance.md

version number. These are the drivers that implement `SyscallDriver` and are
commonly found in capsules. Note, these stabilization guarantees are separate
from the stability of the system call interface itself (i.e., `command`,
`allow`, etc.).

The goal of this stability guarantee is to ensure that applications that use a
particular system call driver will continue to work with a kernel of the same
major version number. However, Tock does not prohibit expanding a stabilized
interface within the same major version number. This enables new functionality
to be added, or for bug fixes to be added without breaking backwards
compatibility.

### Syscall Driver Stabilization Process

The general process is a syscall driver, identified by its driver number, is
proposed to be stabilized. The driver must have a documented interface in the
`docs/syscalls` directory. The interface has a waiting period (which can have
started in the past) where if no changes have been made to the interface the
interface is marked as stable as of the next release of Tock.

Syscall driver stabilization process:

1. The driver has complete documentation in the `doc/syscalls` directory.

> 1. A Tock developer proposes that a specific driver, as identified by its driver
>    number and source code in the upstream Tock repository, should be stabilized.
> 1. The core working group must support the stabilization. If there is support
>    the syscall driver must be moved to the `capsules/core` crate and the
>    stabilization column in `doc/syscalls/README.md` is marked with a ⏰.

Reviewer: ```suggestion
1. A Tock developer proposes that a specific driver, as identified by its driver
   number and source code in the upstream Tock repository, should be stabilized
   by creating a pull request that moves the driver to the `capsules/core` crate and
   marks the stabilization column in `doc/syscalls/README.md` with a ⏰.
1. Stabilization PRs are always considered `P-Significant`, which requires the core
   working group to support the stabilization.
```

version number. These are the drivers that implement `SyscallDriver` and are
commonly found in capsules. Note, these stabilization guarantees are separate
from the stability of the system call interface itself (i.e., `command`,
`allow`, etc.).

The goal of this stability guarantee is to ensure that applications that use a
particular system call driver will continue to work with a kernel of the same
major version number. However, Tock does not prohibit expanding a stabilized
interface within the same major version number. This enables new functionality
to be added, or for bug fixes to be added without breaking backwards
compatibility.

### Syscall Driver Stabilization Process

The general process is a syscall driver, identified by its driver number, is
proposed to be stabilized. The driver must have a documented interface in the
`docs/syscalls` directory. The interface has a waiting period (which can have
started in the past) where if no changes have been made to the interface the
interface is marked as stable as of the next release of Tock.

Syscall driver stabilization process:

1. The driver has complete documentation in the `doc/syscalls` directory.

> 1. A Tock developer proposes that a specific driver, as identified by its driver

Reviewer 2: ```suggestion
2. A Tock developer proposes that a specific driver, as identified by its driver
```

> 1. A Tock developer proposes that a specific driver, as identified by its driver
>    number and source code in the upstream Tock repository, should be stabilized
>    by creating a pull request that moves the driver to the `capsules/core` crate and
>    marks the stabilization column in `doc/syscalls/README.md` with a ⏰.
> 1. Stabilization PRs are always considered `P-Significant`, which requires the core
>    working group to support the stabilization.
> 1. The syscall driver must be unchanged for a period of four months. This period
>    may start from a point before the stabilization process started. The driver
>    must be reasonably tested during this period.
> 1. If changes are required the waiting period resets.
> 1. After the probationary period, the syscall driver is marked stabilized at the

Reviewer 2: ```suggestion
2. A Tock developer proposes that a specific driver, as identified by its driver
   number and source code in the upstream Tock repository, should be stabilized
   by creating a pull request that moves the driver to the `capsules/core` crate and
   marks the stabilization column in `doc/syscalls/README.md` with a ⏰.
3. Stabilization PRs are always considered `P-Significant`, which requires the core
   working group to support the stabilization.
4. The syscall driver must be unchanged for a period of four months. This period
   may start from a point before the stabilization process started. The driver
   must be reasonably tested during this period.
5. If changes are required the waiting period resets.
1. After the probationary period, the syscall driver is marked stabilized at the
```

Author: I'm not sure why I went with the just use "1" in this case, but I don't think we are consistent either way. Do you think we should change to correct the numbering?

Reviewer: A very casual perusal, i.e., `[ ~/code/helena-project/tock] $ git grep -A5 '^1\.'`

suggests that the majority of markdown in the Tock repo uses "accurate" numbering; seems worth aligning with that norm 🤷🏼
